python-client/giskard/datasets/base/__init__.py

```diff
@@ -659,16 +659,27 @@ def copy(self):
     def to_mlflow(self, mlflow_client: MlflowClient = None, mlflow_run_id: str = None):
         import mlflow
 
-        with tempfile.NamedTemporaryFile(prefix="dataset-", suffix=".csv") as f:
+        # To avoid file being open in write mode and read at the same time,
+        # First, we'll write it, then make sure to remove it
+        with tempfile.NamedTemporaryFile(
+            prefix="dataset-", suffix=".csv", delete=False
+        ) as f:
+            # Get file path
             local_path = f.name
-            artifact_name = local_path.split("/")[-1]
-            with open(local_path, "wb") as fw:
-                uncompressed_bytes = save_df(self.df)
-                fw.write(uncompressed_bytes)
+            # Get name from file
+            artifact_name = Path(f.name).name
+            # Write the file on disk
+            f.write(save_df(self.df))
+        try:
             if mlflow_client is None and mlflow_run_id is None:
                 mlflow.log_artifact(local_path)
             elif mlflow_client and mlflow_run_id:
                 mlflow_client.log_artifact(mlflow_run_id, local_path=local_path)
+            # Note(Bazire): Should we have a raising else here ?
```

Comment:
I'd say yes, to make sure there's no case where one of client or run_id are None
WDYT @rabah-khalek